kedro/runner/runner.py

```diff
@@ -166,25 +180,97 @@ def run_only_missing(
 
         return self.run(to_rerun, catalog, hook_manager)
 
+    @abstractmethod  # pragma: no cover
+    def _get_executor(
+        self, max_workers: int
+    ) -> ThreadPoolExecutor | ProcessPoolExecutor:
+        """Abstract method to provide the correct executor (e.g., ThreadPoolExecutor or ProcessPoolExecutor)."""
+        pass
+
     @abstractmethod  # pragma: no cover
```

Comment:
I'd say yes, because you can still overwrite this when creating a custom Runner. But now it's also easier to create a custom runner, because the _run() method has more in place to get started from.